File: arrays/tcs/arrays.ts
// -------------------------------
// 1. Array declaration and creation
// -------------------------------
function create_array(): number[] {
    const array: number[] = [];

    for (let i = 0; i < 10; i++) {
        const randomNumber = Math.floor(Math.random() * 100) + 1;
        array.push(randomNumber);
    }
    
    console.log("Arreglo inicial:");
    console.log(array);
    
    return array;
}

// -------------------------------
// 2. Traversal and printing
// -------------------------------
function traverse_array(array: number[]): void {
    console.log("\nRecorrido con for clásico:");
    for (let i = 0; i < array.length; i++) {
        console.log(`Índice ${i}: ${array[i]}`);
    }

    console.log("\nRecorrido con for-each:");
    for (const value of array) {
        console.log(value);
    }
}

// -------------------------------
// 3. Modification
// -------------------------------
function modify_array(array: number[]): void {
    let odd_numbers: number[] = [];
    let multiplied_array: number[] = [];

    for (let i = 0; i < array.length; i++) {
        odd_numbers[i] = array[i];

        if (array[i] % 2 !== 0) {
            odd_numbers[i] = 0;
        }
    }

    console.log("\nDespués de cambiar impares por 0:");
    console.log(odd_numbers);

    for (let i = 0; i < array.length; i++) {
        multiplied_array.push(array[i] * i);
    }

    console.log("\nDespués de multiplicar por su índice:");
    console.log(multiplied_array);
}

// -------------------------------
// 4. Linear search
// -------------------------------
async function linear_search(array: number[]): Promise<void> {
    const readline = await import("readline");

    const rl = readline.createInterface({
        input: process.stdin,
        output: process.stdout
    });

    const question = (query: string) => new Promise<string>(resolve => rl.question(query, resolve));

    const input = await question("\nIngrese un valor a buscar: ");
    const target_value = parseInt(input);

    for (let i = 0; i < array.length; i++) {
        if (array[i] === target_value) {
            console.log(`Valor encontrado en el índice ${i}`);
            rl.close();
            return;
        }
    }

    console.log("Valor no encontrado");
    rl.close();
}

// -------------------------------
// Main function
// -------------------------------
async function main() {
    const array = create_array();
    traverse_array(array);
    await linear_search(array);
    modify_array(array);
}

main();

File: arrays/py/arrays.py
import random

# -------------------------------
# 1. Array declaration and creation
# -------------------------------
def create_array():
    array = [random.randint(1, 100) for _ in range(10)]
    print("Arreglo inicial:")
    print(array)
    return array

# -------------------------------
# 2. Traversal and printing
# -------------------------------
def traverse_array(array):
    print("\nRecorrido con for:")
    for i in range(len(array)):
        print(f"Índice {i}: {array[i]}")

    print("\nRecorrido con for-each:")
    for value in array:
        print(value)

# -------------------------------
# 3. Modification
# -------------------------------
def modify_array(array):
    odd_numbers = []
    multiplied_array = []

    for i in range(len(array)):
        odd_numbers.append(array[i])
        if array[i] % 2 != 0:
            odd_numbers[i] = 0

    print("\nDespués de cambiar impares por 0:")
    print(odd_numbers)

    for i in range(len(array)):
        multiplied_array.append(array[i] * i)

    print("\nDespués de multiplicar por su índice:")
    print(multiplied_array)

# -------------------------------
# 4. Linear search
# -------------------------------
def linear_search(array):
    target_value = int(input("\nIngrese un valor a buscar: "))
    
    for i in range(len(array)):
        if array[i] == target_value:
            print(f"Valor encontrado en el índice {i}")
            return
    
    print("Valor no encontrado")




def main():
    array = create_array()
    traverse_array(array)
    linear_search(array)   
    modify_array(array)

if __name__ == "__main__":
    main()